superset-frontend/src/explore/components/ExploreChartHeader/index.jsx

```diff
@@ -157,13 +157,26 @@ export class ExploreChartHeader extends React.PureComponent {
 
           if (dashboard && dashboard.json_metadata) {
             // setting the chart to use the dashboard custom label colors if any
-            const labelColors =
-              JSON.parse(dashboard.json_metadata).label_colors || {};
+            const metadata = JSON.parse(dashboard.json_metadata);
+            const sharedLabelColors = metadata.shared_label_colors || {};
+            const labelColors = metadata.label_colors || {};
+
             const categoricalNamespace =
               CategoricalColorNamespace.getNamespace();
 
+            Object.keys(sharedLabelColors).forEach(label => {
+              categoricalNamespace.setColor(
+                label,
+                sharedLabelColors[label],
+                metadata.color_scheme,
+              );
+            });
             Object.keys(labelColors).forEach(label => {
-              categoricalNamespace.setColor(label, labelColors[label]);
+              categoricalNamespace.setColor(
+                label,
+                labelColors[label],
+                metadata.color_scheme,
+              );
```

Comment:
What do you think about merging `sharedLabelColors` and `labelColors` in one map and then only loop through one map? Also, I think it would make sense to change the name of `labelColors` with `customLabelColors` for clarity.

Reply:
Great. I agree to change the name to `customLabelColors` but should we need to change the fields on the backend?  Or just changing the variable name here

Reply:
If that helps bring more clarity to the backend code, I'd say yes. However, I am happy with changing it on the frontend only for now